hnn_core/viz.py

```diff
@@ -609,5 +655,89 @@ def plot_connectivity_matrix(conn, ax=None, show=True):
     ax.set_title(f"{conn['src_type']} -> {conn['target_type']} "
                  f"({conn['loc']}, {conn['receptor']})")
 
+    plt.tight_layout()
     plt_show(show)
     return ax.get_figure()
+
+
+def plot_cell_connectivity(net, conn_idx, src_gid, colorbar=True,
+                           ax=None, show=True):
+    """Plot synaptic weight of connections from a src_gid.
+
+    Parameters
+    ----------
+    net : Instance of Network object
+        The Network object
+    conn_idx : int
+        Index of connection to be visualized
+        from `net.connectivity`
+    src_gid : int
+        Each cell in a network is uniquely identified by it's "global ID": GID.
+    ax : instance of Axes3D
+        Matplotlib 3D axis
+    show : bool
+        If True, show the plot
+
+    Returns
+    -------
+    fig : instance of matplotlib Figure
+        The matplotlib figure handle.
+
+    """
+    import matplotlib.pyplot as plt
+    from .network import Network
+    from .cell import _get_gaussian_connection
```

Comment:
don't nest this if possible ... not sure if you have a circular import issue

Reply:
Just checked, unfortunately the nesting is required here as well.